airbyte-integrations/connectors/source-google-workspace-admin-reports/source_google_workspace_admin_reports/api.py

```diff
@@ -111,7 +117,44 @@ def read(self, getter: Callable, params: Dict = None) -> Iterator:
                 break
 
 
-class ActivitiesAPI(StreamAPI):
+class IncrementalStreamAPI(StreamAPI, ABC):
+    """Stream that supports state and incremental read"""
+
+    state_pk = "time"
+
+    @property
+    def state(self) -> Optional[Mapping[str, Any]]:
+        """Current state, if wasn't set return None"""
+        if self._state:
+            return {self.state_pk: str(self._state)}
+        return None
+
+    @state.setter
+    def state(self, value):
+        self._state = pendulum.parse(value[self.state_pk])
+        self._start_time = self._state.isoformat()
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._state = None
+
+    def read(self, getter: Callable, params: Mapping[str, Any] = None) -> Iterator:
+        """Update cursor(state)"""
+        params = params or {}
+        cursor = None
+        for record in super().read(getter, params):
+            "Report API return records from newest to oldest"
+            if not cursor:
+                cursor = pendulum.parse(record[self.state_pk])
+            yield record
+
+        if cursor:
+            new_state = max(cursor, self._state) if self._state else cursor
+            if new_state != self._state:
+                self._state = new_state
```

Comment:
could you add a log statement like `f"advancing cursor value for {self.name} stream from X to Y"` ?